### PR TITLE
resource_control: Introduce resource group priority 

### PR DESCRIPTION
**Pull request by Connor1996**
### What is changed and how it works?

Issue Number: Ref https://github.com/tikv/tikv/issues/13730

What's Changed:

```commit-message
Introduce resource group priority, tasks are scheduled based on the order of (priority, virtual_time)
```

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Introduce resource group priority
```

### CODE REVIEW

File: components/resource_control/src/resource_group.rs [hunk 1]
```diff
@@ -282,9 +297,22 @@ impl TaskPriorityProvider for ResourceController {
     }
 }
 
+fn concat_priority_vt(mut group_priority: u32, vt: u64) -> u64 {
+    assert!(group_priority <= 16);
+    // map 0 to medium priority(default priority)
+    if group_priority == 0 {
```
Comment (glorv) [inline components/resource_control/src/resource_group.rs:303]: I prefer to move this adjust and (maybe) bound check add `add_resource_group`

File: components/resource_control/src/channel.rs [hunk 2]
```diff
@@ -87,39 +87,42 @@ impl<T: Send + 'static> Sender<T> {
...
-                    low_bound,
-                );
-                sender.try_send(m, priority).map(|_| priority)
+                let p = resource_ctl
+                    .get_priority(last_msg_group.borrow().as_bytes(), CommandPri::Normal);
+                let priority = if let Some(low_bound) = low_bound {
+                    std::cmp::max(p, low_bound)
+                } else {
+                    p
+                };
```
Comment (glorv) [inline components/resource_control/src/channel.rs:124]:
```suggestion
                let priority = std::cmp::max(p, low_bound.unwrap_or(0));
```

**Review by glorv — COMMENTED** [review]
LGTM

**Review by glorv — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by nolouch** [comment]
@Connor1996  Here are some conflicts.

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**